packages/astro/src/content/runtime.ts

```diff
@@ -256,6 +265,14 @@ type DataEntryResult = {
 
 type EntryLookupObject = { collection: string; id: string } | { collection: string; slug: string };
 
+function emulateLegacyEntry({ legacyId, id, ...entry }: DataEntry) {
```

Comment:
this is a lot of object allocations in this function. isn't the `...entry` here already creating a new object, and if so why not just append the new props to it and return it? Maybe this is a nit and doesn't matter much, I just hate seeing object allocations just to make the types slightly easier.